sqld/src/database/config.rs

```diff
@@ -0,0 +1,84 @@
+use parking_lot::Mutex;
+use serde::{Deserialize, Serialize};
+use std::path::{Path, PathBuf};
+use std::sync::Arc;
+use std::{fs, io};
+
+use crate::error::Error;
+use crate::Result;
+
+pub struct DatabaseConfigStore {
+    config_path: PathBuf,
+    tmp_config_path: PathBuf,
+    config: Mutex<Arc<DatabaseConfig>>,
```

Comment:
I don't know, `RwLock` would allow us to clone the `Arc` concurrently, but isn't the overhead of `RwLock` higher than `Mutex`? The cloning of the `Arc` is be so fast that there should be almost no contention on this lock anyway.

Reply:
uhm yeah maybe it doesn't matter... it'll probably get unlocked during the spin-locking phase anyway